modules/ballerina-http/src/main/java/org/ballerinalang/net/http/nativeimpl/request/GetFormParams.java

```diff
@@ -52,20 +53,22 @@
     public BValue[] execute(Context context) {
         try {
             BStruct requestStruct  = ((BStruct) getRefArgument(context, 0));
-            //TODO check below line
             HTTPCarbonMessage httpCarbonMessage = HttpUtil
                     .getCarbonMsg(requestStruct, HttpUtil.createHttpCarbonMessage(true));
+
             String contentType = httpCarbonMessage.getHeader(Constants.CONTENT_TYPE);
             if (contentType != null && contentType.contains(Constants.APPLICATION_FORM)) {
                 String payload;
-                if (httpCarbonMessage.isAlreadyRead()) {
-                    payload = httpCarbonMessage.getMessageDataSource().getMessageAsString();
+                if (requestStruct.getNativeData(Constants.MESSAGE_DATA_SOURCE) != null) {
+                    MessageDataSource messageDataSource = (MessageDataSource) requestStruct
+                            .getNativeData(Constants.MESSAGE_DATA_SOURCE);
```

Comment:
Shall we have a common method for extracting the MESSAGE_DATA_SOURCE from the struct? Something like below may be in HttpUtil class which can used in all the places where MESSAGE_DATA_SOURCE is extracted from struct .
`
public static MessageDataSource getMessageDatasource(BStruct requestStruct) {
        MessageDataSource source = null;
        if (requestStruct.getNativeData(Constants.MESSAGE_DATA_SOURCE) != null) {
            source = (MessageDataSource) requestStruct.getNativeData(Constants.MESSAGE_DATA_SOURCE);
        }
        return source;
    }
`